graphql_compiler/compiler/ir_lowering_gremlin.py

```diff
@@ -148,11 +148,13 @@ def validate(self):
             raise TypeError(u'Expected FoldScopeLocation fold_scope_location, got: {} {}'.format(
                 type(self.fold_scope_location), self.fold_scope_location))
 
-        allowed_block_types = (GremlinFoldedFilter, Traverse, Backtrack)
+        allowed_block_types = (GremlinFoldedFilter, GremlinFoldedTraverse, Backtrack)
         for block in self.folded_ir_blocks:
             if not isinstance(block, allowed_block_types):
-                raise AssertionError(u'Found non-GremlinFoldedFilter in folded_ir_blocks: {} {} '
-                                     u'{}'.format(type(block), block, self.folded_ir_blocks))
+                raise AssertionError(u'Found invalid block of type {} in folded_ir_blocks: {} '
+                                     u'Allowed types are {}. '
+                                     .format(type(block), self.folded_ir_blocks,
+                                             allowed_block_types))
```

Comment:
try to avoid formatting into these "jagged" formations -- you can move the error message string to a new line if need be:
```
raise AssertionError(
    u'error message '
    u'here: {}'.format(stuff))
```